extensions/mssql/src/sparkFeature/dialog/dialogCommands.ts

```diff
@@ -49,31 +55,61 @@ export class OpenSparkJobSubmissionDialogCommand extends Command {
 
 	private async selectConnection(): Promise<SqlClusterConnection> {
 		let connectionList: azdata.connection.Connection[] = await this.apiWrapper.getActiveConnections();
-		let displayList: string[] = new Array();
 		let connectionMap: Map<string, azdata.connection.Connection> = new Map();
+		let selectedHost: string = undefined;
+		let showConnectionDialog = false;
+
+		// Filter invalid connections
+		if (connectionList && connectionList.length > 0) {
+			connectionList = connectionList.filter(conn => conn.providerName === constants.sqlProviderName && (<MssqlOptions><any>conn.options).server);
+		}
+		// Prompt choice if we have active connections
 		if (connectionList && connectionList.length > 0) {
-			connectionList.forEach(conn => {
-				if (conn.providerName === constants.sqlProviderName) {
-					displayList.push(conn.options.host);
-					connectionMap.set(conn.options.host, conn);
-				}
+			let selectConnectionMsg = localize('selectOtherServer', 'Select other SQL Server');
+			let displayList: string[] = [];
+			if (connectionList && connectionList.length > 0) {
```

Comment:
this is already true, we are in this condition scope already.

Reply:
Thanks - missed in a refactor